torchtune/models/clip/_text_encoder.py

```diff
@@ -87,10 +90,9 @@ def forward(
             )
         x = self.final_norm(x)
 
-        # Select the output of the EOS token for each encoding in the batch
+        # Select the output of the EOT token for each encoding in the batch
         # [b, s, d] -> [b, d]
-        # TODO: handle the case when the EOS token is not the highest token ID
-        eos_token_positions = tokens.argmax(dim=-1)
+        eos_token_positions = (tokens == self.eot_token).int().argmax(dim=-1)
```

Comment:
can do `(tokens == self.eot_token).nonzero()`

Reply:
we don't want all positions of the eot token, just the first one (argmax gives first position where they match)